service/src/main/java/bio/terra/cbas/dependencies/wes/CromwellClient.java

```diff
@@ -31,15 +33,30 @@ public CromwellClient(
     singletonHttpClient = new ApiClient().getHttpClient();
   }
 
-  private ApiClient getApiClient()
+  public ApiClient getWriteApiClient()
```

Comment:
Like @dillydally414 did with having thread local beans for the user token, I wonder whether we could have thread-local write api clients? That would mean it would be impossible for us to inappropriately store-and-retrieve the wrong client in a downstream cache (which is a concern that it'd be nice to make impossible rather than having to logic check every time we use a cache)

Reply:
@kpierre13 Is this something you plan to address in the follow up ticket to use the appropriate user's cromwell rather than in this PR?

Reply:
Yes, this work is currently implemented in the next PR

Reply:
I think we could still cache the URLs separately based on user, just not the entire client (which comes pre-loaded with the tokens/authorization)